File: LeetCode/Medium/0148-sort-list/0148-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        hq = []

        while curr:
            heapq.heappush(hq, curr.val)
            curr = curr.next

        curr = head

        while hq:
            curr.val = heappop(hq)
            curr = curr.next
        
        return head

File: LeetCode/Medium/0148-sort-list/0148-sort-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func sortList(head *ListNode) *ListNode {
    count := make([]int, 200001)
    curr := head
    
    for curr != nil {
        count[curr.Val+100000]++
        curr = curr.Next
    }
    
    curr = head
    
    for i := 0; i < 200001; i++ {
        for count[i] != 0 {
            count[i]--
            curr.Val = i - 100000
            curr = curr.Next
        }
    }
    
    return head
}